### ISSUE TITLE
[clang-tidy] False-positive result of check readability-redundant-parentheses with rvalue overload

### ISSUE THREAD

**Issue by pkl97** [llvm/llvm-project#192435]
This code triggers a false-positive result from clang-tidy check readability-redundant-parentheses (see https://godbolt.org/z/bME79eef6):
```c++
#include <sstream>
#include <iostream>

std::string getString()
{
    return (std::ostringstream{} << "Hello World").str(); // <-- this line triggers the warning
}

int main()
{
    std::cout << getString() << '\n';
    return 0;
}
```

The issue occurs on the main branch (upcoming clang 23). In clang 22 there was no such warning.
